framework/src/org/checkerframework/framework/type/TypesIntoElements.java

```diff
@@ -244,7 +244,8 @@ private static void addUniqueTypeCompounds(Types types, Symbol sym, List<TypeCom
         List<Attribute.TypeCompound> res = List.nil();
 
         for (Attribute.TypeCompound tc : tcs) {
-            if (!TypeAnnotationUtils.isTypeCompoundContained(types, raw, tc)) {
+            // Don't insert an annotation if one with the same name already exists.
+            if (!TypeAnnotationUtils.isTypeCompoundContainedByName(types, raw, tc)) {
```

Comment:
Because the Lock Checker can't differentiate between annotations in elements that have ben canonicalized and those that haven't, canonicalization is always performed by annotateImplicit(Tree, ATM) or annotateImplicit(Element, ATM).  So, it doesn't matter if the annotation in the element is changed to the canonicalized version.  I think it makes more sense to write out the annotation that was written.